tests/Http/Controllers/Api/GeoOverlayControllerTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Biigle\Tests\Modules\Geo\Http\Controllers\Api;
+
+use ApiTestCase;
+use Biigle\Tests\Modules\Geo\GeoOverlayTest;
+use Storage;
+
+class GeoOverlayControllerTest extends ApiTestCase
+{
+    public function testShowFile()
+    {
+        Storage::fake('geo-overlays');
+        $overlay = GeoOverlayTest::create();
+        $overlay->volume_id = $this->volume()->id;
+        $overlay->save();
+        $id = $overlay->id;
+        Storage::disk('geo-overlays')->put($overlay->path, 'content');
+
+        $this->doTestApiRoute('GET', "/api/v1/geo-overlays/{$id}/file");
+
+        $this->beUser();
+        $this->get("/api/v1/geo-overlays/{$id}/file")
+            ->assertStatus(403);
+
+        $this->beGuest();
+        $response = $this->json('GET', "/api/v1/geo-overlays/{$id}/file")
+            ->assertStatus(200);
+        $this->assertEquals(7, $response->headers->get('content-length'));
+    }
+
+    // ToDo: Find out why get this error --> Unable to retrieve the file_size for file at location
+    // 
```

Comment:
Hi @mzur, for this test I get an error `Unable to retrieve the file_size for file at location: 2/2_original.`. This is correct, as the file should not exist (since it wasn't created beforehand). I would, however, expect that the GeoOverlayController would return a `404 FileNotFoundException`. I didn't find an answer to why this error occurs, but this [link](https://stackoverflow.com/questions/73074192/unable-to-retrieve-the-file-size-for-file-when-using-storagefake-for-testing) suggests it might be due to an upgrade from Laravel version 8 to 9?

Reply:
When I run the test as it is here, it works. The file is created in line 18, too. But if I remove that line, I get your error.

The simple answer is that the `FileNotFoundException` is no longer what needs to be caught in the controller if the file is missing. Now you have to catch `League\Flysystem\UnableToRetrieveMetadata`. This has likely changed because of the Laravel upgrade you mention (during which the geo overlay code was already removed).

Reply:
Thank you for pointing that out!

Reply:
resolved c7cb2b0b2b77ba3bd001664069c384d7e6e8007e